python/sdk/test/requirements_test.py

```diff
@@ -0,0 +1,168 @@
+import pytest
+import yaml
+from merlin.requirements import process_conda_env
+
+
+@pytest.mark.parametrize(
+    "input,output,additional_merlin_reqs",
+    [
+        # Test using file
+        (
+            "test/requirements/empty_in.yaml",
```

Comment:
can we do everything in code? There are a lot of files and isn't that big

@deadlycoconuts opinion?

Reply:
not everything, though. we still need to have requirements.txt to be available so it can be referred and parsed.
imo, i found it easier to write and read these test fixtures (these .yaml files) instead of writing the test object in python code.

Reply:
Hmm I actually like how we're testing using actual files though, given that the function `process_conda_env` being tested is intended to work for both 'a dictionary representation of a conda environment or the path to a conda environment yaml file'. Otherwise, if the test cases to test the processing of a conda env yaml file are to be written in code, are you saying we should preprocess them into yaml files just before `process_conda_env` is called in `test_process_conda_env`?